modules/mini-games/Race.lua

```diff
@@ -260,25 +264,42 @@ local function Nth (n) return n..getSuffix(n) end
 
 --- Function called by mini game module to stop a race
 local function stop()
-    -- Print the place that each player came
     local results = {}
-    for name, value in pairs(scores["finish_times"]) do
-        local time = value[2]
-
-        local up_result = results[#results]
-        if up_result and up_result.score == math.round(time, 2) then
-            up_result.players[#up_result.players + 1] = name
-
-        else
-            results[#results + 1] = {
-                place = value[1],
-                score = math.round(time, 2),
-                players = {name}
-            }
+    local results_by_name = {}
+
+    -- Add scores for players that finished the game
+    for name, result in pairs(variables["finish_times"]) do
+        results[#results + 1] = {
+            place = result.place,
+            score = math.round(result.time, 2),
+            players = {name}
+        }
+        results_by_name[name] = results[#results]
+
+        -- Correct placement in case of score ties
+        if #results > 1 then
+            local prev = results[#results - 1]
+            local curr = results[#results]
+            if curr.score == prev.score then
+                curr.place = prev.place
+            end
+        end
+    end
```

Comment:
`#results` is used alot here, I feel it would be better to include an incremental variable, or to init the value of curr as the table rather than setting it in the table directly. ie
```lua
-- Outside the loop
local last = 0
-- Inside the loop
local curr = { ... }
last = last + 1
results[last] = curr
results_by_name[name] = curr
if last > 1 then
    local prev = results[last - 1]
    -- ...
end
```

Reply:
Implemented your suggestion.